packages/reporters/dev-server/src/HMRServer.js

```diff
@@ -272,9 +273,19 @@ export default class HMRServer {
   }
 
   broadcast(msg: HMRMessage) {
-    const json = JSON.stringify(msg);
-    for (let ws of this.wss.clients) {
-      ws.send(json);
+    if (msg.type === 'update' && msg.assets.length > BROADCAST_MAX_ASSETS) {
+      // Split up message if too large
```

Comment:
FYI splitting could result in different behavior in the browser. The logic in hmr-runtime.js does various checks over all assets in a message. Maybe it's ok just pointing it out in case.

Reply:
I didn't bother looking into if it worked correctly and just refactored this by introducing a partial message and gets joined together. Hopefully that preserves the current behaviour sufficently.